captum/optim/_core/loss.py

```diff
@@ -8,37 +10,197 @@
 from captum.optim._utils.typing import ModuleOutputMapping
 
 
+def _make_arg_str(arg):
+    arg = str(arg)
+    too_big = len(arg) > 15 or "\n" in arg
+    return arg[:15] + "..." if too_big else arg
+
+
 class Loss(ABC):
     """
     Abstract Class to describe loss.
+    Note: All Loss classes should expose self.target for hooking by
+    InputOptimization
     """
 
-    def __init__(self, target: nn.Module) -> None:
+    def __init__(self) -> None:
         super(Loss, self).__init__()
-        self.target = target
+
+    @abstractproperty
+    def target(self):
+        pass
 
     @abstractmethod
-    def __call__(self, targets_to_values: ModuleOutputMapping):
+    def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
         pass
 
+    def __repr__(self):
+        return self.__name__
+
+    def __neg__(self):

```

Comment:
Is it possible to add a return type to all these math op functions and the `__repr__` function?